File: src/test/java/shopping/wish/domain/WishlistTest.java
package shopping.wish.domain;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class WishlistTest {
    @Test
    @DisplayName("위시리스트를 만들면 활성 상태로 시작한다")
    void createInitializeActiveWishlist() {
        Wishlist wishlist = Wishlist.create(1L);

        assertThat(wishlist.getMemberId()).isEqualTo(1L);
        assertThat(wishlist.getStatus()).isEqualTo(WishlistStatus.ACTIVE);
        assertThat(wishlist.isActive()).isTrue();
    }

    @Test
    @DisplayName("위시리스트를 삭제하면 비활성 상태가 된다")
    void deleteMarkWishlistDeleted() {
        Wishlist wishlist = Wishlist.create(1L);

        wishlist.delete();

        assertThat(wishlist.getStatus()).isEqualTo(WishlistStatus.DELETED);
        assertThat(wishlist.isActive()).isFalse();
        assertThat(wishlist.getDeletedAt()).isNotNull();
    }
}

File: src/test/kotlin/shopping/wish/domain/WishListTest.kt
package shopping.wish.domain

import io.kotest.assertions.throwables.shouldNotThrowAny
import io.kotest.core.spec.style.BehaviorSpec
import shopping.wish.fixture.WishListFixture

class WishListTest : BehaviorSpec({
    Given("위시 리스트의 상품 목록들을") {
        val wishList = WishListFixture.`고객 1의 위시 리스트`.`엔티티 생성`()

        When("적절한 컬렉션으로 변환하여") {
            shouldNotThrowAny {
                wishList.getWishProducts()
            }
        }
    }
})
